### PR TITLE
Convert fx.rst to fx.md

### LINKED ISSUE #155023: Convert to markdown: fx.rst, hub.rst, jit_builtin_functions.rst, jit_language_reference_v2.rst, jit_language_reference.rst
Convert the following files to MyST markdown and ensure that the doc tests are green:

- [fx.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/fx.rst)
- [hub.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/hub.rst)
- [jit_builtin_functions.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/jit_builtin_functions.rst)
- [jit_language_reference_v2.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/jit_language_reference_v2.rst)
- [jit_language_reference.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/jit_language_reference.rst)

To convert an `.rst` file:

1) Assign the issue to yourself using the `/assigntome` comment.

2) **Important!** Use `git mv filename.rst filename.md` to convert the issue from `.rst` to `.md`. **DO NOT delete**
   the `.rst` and create the `.md` - we won't accept PRs with deleted `.rst` files. We want to preserve the git log of
   those files. Therefore, using `git mv` is very important!

3) Before submitting a PR, build the docs locally as described in the [README](https://github.com/pytorch/pytorch/blob/main/README.md#building-the-documentation).
   Preview localy with `make serve` to ensure that the HTML renders correctly.    

4) Run `lintrunner` from the root of the repository:

   ```
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

4) Submit a PR with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:

   * pull / linux-docs / build-docs-python-false (pull_request)
   * linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)

   Doc tests start after the Docker build, which might take about 30 minutes. 
   The doc build itself should take about 25 minutes. When the build is done, 
   you can see the preview of your build by clicking:
   📄 **Preview Python docs built from this PR** in the `pytorch-bot` comment.

   Make sure your PR is tagged with `docathon-h1-2025`, `module: docs`, `topic: not user facing`,
   and `topic: docs` labels.

5) Compare the preview of the pages you converted with same pages it the main branch.
   The converted pages in preview must look the same as in the main branch.

6) Once the PR is accepted, one of the maintainers will merge it.

See [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#restructuredtext-vs-markdown)
in our wiki or [MyST Parser documentation](https://myst-parser.readthedocs.io/en/v0.15.1/sphinx/use.html) for detailed syntax.

Here is an example [PR](https://github.com/pytorch/pytorch/pull/154438)

### PR DESCRIPTION
**Pull request by Rachel0619**
Fixes #155023

@pytorchbot label "topic: docs"
@pytorchbot label "topic: not user facing"
@pytorchbot label docathon-h1-2025
@pytorchbot label module: docs

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

**Comment by svekars** [comment]
We need a git mv of fx.rst => fx.md. Can you please make sure the fx.rst is part of this PR.

**Comment by Rachel0619** [comment]
That would exceed the 2000 lines of limit for the sanity check. Any suggestions for a workaround?

**Comment by svekars** [comment]
can do something like this: https://github.com/pytorch/pytorch/pull/155492 - feel free to grab that PR - this avoids moving code blocks and therefore the number of changes is smaller. Then, plan to submit a follow up PR to move properly fix indentation in those code blocks.

**Comment by Rachel0619** [comment]
grabbed the PR #155492 from @svekars 
Big thanks for helping out!

File: docs/source/fx.md [hunk 1]
```diff
@@ -678,81 +670,78 @@ Going off of the example above, consider the following code:
...
+to `transform_graph(traced)`.
 
-We may also have good luck by editing the ``print_tabular`` method to print
+We may also have good luck by editing the `print_tabular` method to print
 different attributes of the Nodes in the Graph. (For example, we might
-want to see the Node’s ``input_nodes`` and ``users``.)
+want to see the Node’s `input_nodes` and `users`.)
 
-.. _Available Debuggers:
+(Available Debuggers)=
```
Comment (svekars) [inline docs/source/fx.md:685]:
```suggestion
(available_debuggers)=
```

File: docs/source/fx.md [hunk 2]
```diff
@@ -421,33 +421,31 @@ disprove correctness. Then, inspect and debug the generated code. Then, debug th
 process of transformations that led to the generated code.
 
 If you’re not familiar with debuggers, please see the auxiliary section
-:ref:`Available Debuggers`.
+{ref}`Available Debuggers`.
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
{ref}`Available-Debuggers`.
```

File: docs/source/fx.md (hunk 1 above)
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
(Available-Debuggers)=
```

File: docs/source/fx.md [hunk 3]
```diff
@@ -572,38 +566,36 @@ your code and examine it from there.
...
+```
 After running the above example, we can then look at the code within
-``foo/module.py`` and modify it as desired (e.g. adding ``print``
-statements or using ``pdb``) to debug the generated code.
+`foo/module.py` and modify it as desired (e.g. adding `print`
+statements or using `pdb`) to debug the generated code.
 
-Debugging the Transformation
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#### Debugging the Transformation
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
### Debugging the Transformation
```

File: docs/source/fx.md [hunk 4]
```diff
@@ -207,212 +211,208 @@ can be found below.
...
--  `Shape
-   Propagation <https://github.com/pytorch/pytorch/blob/master/torch/fx/passes/shape_prop.py>`__
--  `Performance Profiler <https://github.com/pytorch/tutorials/pull/1319>`__
+-  [ShapePropagation](https://github.com/pytorch/pytorch/blob/master/torch/fx/passes/shape_prop.py)
+-  [Performance Profiler](https://github.com/pytorch/tutorials/pull/1319)
 
 
-Debugging
------------
+### Debugging
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
## Debugging
```

File: docs/source/fx.md [hunk 5]
```diff
@@ -207,212 +211,208 @@ can be found below.
...
 get a concrete output value from the Module, you would call the
-:meth:`Transformer.transform` method to return a new
-:class:`GraphModule` which was subject to any transformation rules
+{meth}`Transformer.transform` method to return a new
+{class}`GraphModule` which was subject to any transformation rules
 you installed as overridden methods.
 
-Examples of the Interpreter Pattern
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+## Examples of the Interpreter Pattern
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
#### Examples of the Interpreter Pattern
```

File: docs/source/fx.md [hunk 6]
```diff
@@ -207,212 +211,208 @@ can be found below.
...
+-  [Performance Profiler](https://github.com/pytorch/tutorials/pull/1319)
 
 
-Debugging
------------
+### Debugging
 
-Introduction
-^^^^^^^^^^^^^^^^
+#### Introduction
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
### Introduction
```

File: docs/source/fx.md [hunk 7]
```diff
@@ -421,33 +421,31 @@ disprove correctness. Then, inspect and debug the generated code. Then, debug th
 process of transformations that led to the generated code.
 
 If you’re not familiar with debuggers, please see the auxiliary section
-:ref:`Available Debuggers`.
+{ref}`Available Debuggers`.
 
 
-Common Pitfalls in Transform Authoring
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#### Common Pitfalls in Transform Authoring
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
### Common Pitfalls in Transform Authoring
```

File: docs/source/fx.md [hunk 8]
```diff
@@ -421,33 +421,31 @@ disprove correctness. Then, inspect and debug the generated code. Then, debug th
...
-  as of Python 3.7 (and as of cPython 3.6). A ``dict`` can be used equivalently
-  to a set by storing values to be deduplicated in the keys of the ``dict``.
+  The recommended alternative is to use a `dict` data type, which is
+  [insertion ordered](https://mail.python.org/pipermail/python-dev/2017-December/151283.html)
+  as of Python 3.7 (and as of cPython 3.6). A `dict` can be used equivalently
+  to a set by storing values to be deduplicated in the keys of the `dict`.
 
-Checking Correctness of Modules
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#### Checking Correctness of Modules
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
### Checking Correctness of Modules
```

File: docs/source/fx.md [hunk 9]
```diff
@@ -474,38 +472,35 @@ example:
...
+```python
     assert torch.allclose(resnet18(input_image), transformed_resnet18(input_image))
-
+```
 This is the first tool in our toolbox to check if transformed modules are
 behaving as we expect compared to a reference implementation.
 
-Debugging the Generated Code
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#### Debugging the Generated Code
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
### Debugging the Generated Code
```

File: docs/source/fx.md [hunk 10]
```diff
@@ -474,38 +472,35 @@ example:
...
+traditional debugging techniques like `print` statements or `pdb` is
 not as straightforward. Luckily, we have several techniques we can use
 for debugging the generated code.
 
-Use ``pdb``
-~~~~~~~~~~~~~
-Invoke ``pdb`` to step into the running program. Although the code that
-represents the :class:`Graph` is not in any source file, we can still step
-into it manually using ``pdb`` when the forward pass is invoked.
+## Use `pdb`
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
#### Use `pdb`
```

File: docs/source/fx.md [hunk 11]
```diff
@@ -530,17 +525,16 @@ into it manually using ``pdb`` when the forward pass is invoked.
...
 
     my_module_transformed(input_value)
+```
+(Print the Generated Code)=
 
-.. _Print the Generated Code:
-
-Print the Generated Code
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
+## Print the Generated Code
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
#### Print the Generated Code
```

File: docs/source/fx.md [hunk 12]
```diff
@@ -572,38 +566,36 @@ your code and examine it from there.
     # now compare the output of both the original and the traced version.
     pre_trace = M()
     post_trace = SubclassM()
-
-Use the ``to_folder`` Function From ``GraphModule``
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-:meth:`GraphModule.to_folder` is a method in ``GraphModule`` that allows
+```
+## Use the `to_folder` Function From `GraphModule`
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
#### Use the `to_folder` Function From `GraphModule`
```

File: docs/source/fx.md [hunk 13]
```diff
@@ -678,81 +670,78 @@ Going off of the example above, consider the following code:
...
 different attributes of the Nodes in the Graph. (For example, we might
-want to see the Node’s ``input_nodes`` and ``users``.)
+want to see the Node’s `input_nodes` and `users`.)
 
-.. _Available Debuggers:
+(Available Debuggers)=
 
-Available Debuggers
-^^^^^^^^^^^^^^^^^^^^^^
+#### Available Debuggers
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
### Available Debuggers
```

File: docs/source/fx.md [hunk 14]
```diff
@@ -678,81 +670,78 @@ Going off of the example above, consider the following code:
...
 window in your IDE (e.g. View → Terminal in VSCode), or b) use the
-built-in debugger (usually a graphical wrapper around ``pdb``).
+built-in debugger (usually a graphical wrapper around `pdb`).
 
-.. _Limitations of Symbolic Tracing:
+(Limitations of Symbolic Tracing)=
 
-Limitations of Symbolic Tracing
--------------------------------
+### Limitations of Symbolic Tracing
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
## Limitations of Symbolic Tracing
```

File: docs/source/fx.md [hunk 15]
```diff
@@ -678,81 +670,78 @@ Going off of the example above, consider the following code:
...
 **symbolic** in that the data flowing through the program during this
-execution is not real data, but rather symbols (:class:`Proxy` in FX parlance).
+execution is not real data, but rather symbols ({class}`Proxy` in FX parlance).
 
 Although symbolic tracing works for most neural net code, it has some
 limitations.
 
-Dynamic Control Flow
-^^^^^^^^^^^^^^^^^^^^
+#### Dynamic Control Flow
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
### Dynamic Control Flow
```

File: docs/source/fx.md [hunk 16]
```diff
@@ -766,23 +755,22 @@ For example, let’s examine the following program:
...
+```
+The condition to the `if` statement relies on the value of `x.sum()`,
+which relies on the value of `x`, a function input. Since
+`x` can change (i.e. if you pass a new input tensor to the traced
 function), this is *dynamic control flow*. The traceback walks back up
 through your code to show you where this situation happens.
 
-Static Control Flow
-~~~~~~~~~~~~~~~~~~~
+## Static Control Flow
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
### Static Control Flow
```

File: docs/source/fx.md [hunk 17]
```diff
@@ -1072,48 +1056,57 @@ Miscellanea
...
-
-  - Because of this difference, consider marking modules that interact with the ``training`` flag dynamically as leaf modules.
+    ```
+  - Because of this difference, consider marking modules that interact with the `training` flag dynamically as leaf modules.
 
 
-API Reference
--------------
-
+#### API Reference
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
## API Reference
```

File: docs/source/fx.md [hunk 18]
```diff
@@ -474,38 +472,35 @@ example:
     """
```
Comment (svekars) [inline docs/source/fx.md:463]: ah, missed the closing backticks here for the code block

**Comment by Rachel0619** [comment]
Just fixed some syntax errors that caused warnings. 751a7f8319411c813db0c7557946f69a5e15e0d7

- changed `{eval_rst}` to `{eval-rst}`
- added `__init__` to `automethod`
- added missing closing backticks for the code block

There's still a warning on my end when building locally: 

> /workspaces/pytorch/docs/source/export.rst:764: WARNING: toctree contains reference to nonexisting document 'generated/exportdb/index'

I'm not quite sure whether the root cause is on my end. Might need some help here, thank you!

File: docs/source/fx.md [hunk 19]
```diff
@@ -421,33 +412,31 @@ disprove correctness. Then, inspect and debug the generated code. Then, debug th
...
+### Common Pitfalls in Transform Authoring
 
-* Nondeterministic ``set`` iteration order. In Python, the ``set`` datatype is
-  unordered. Using ``set`` to contain collections of objects like ``Node``\ s,
+* Nondeterministic `set` iteration order. In Python, the `set` datatype is
+  unordered. Using `set` to contain collections of objects like `Node`\ s,
   for example, can cause unexpected nondeterminism. An example is iterating
-  over a set of ``Node``\ s to insert them into a ``Graph``. Because the
-  ``set`` data type is unordered, the ordering of the operations in the output
+  over a set of `Node`\ s to insert them into a `Graph`. Because the
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
  over a set of `Node` s to insert them into a `Graph`. Because the
```

File: docs/source/fx.md [hunk 20]
```diff
@@ -843,24 +823,23 @@ during symbolic tracing:
...
 In the case of truly dynamic control flow, the sections of the program
 that contain this code can be traced as calls to the Method (see
-:ref:`Customizing Tracing`) or function (see
-:func:`wrap`) rather than tracing through them.
+{ref}`Customizing Tracing`) or function (see
+{func}`wrap`) rather than tracing through them.
 
-Non-\ ``torch`` Functions
-^^^^^^^^^^^^^^^^^^^^^^^^^
+### Non-\ `torch` Functions
```
Comment (svekars) [inline docs/source/fx.md]:
```suggestion
### Non- `torch` Functions
```

**Review by svekars — APPROVED** [review]
Ok, this looks good, after this PR is merged please submit a follow up to fix indentation in code blocks.

**Comment by Rachel0619** [comment]
Sure, I'll work on that later. Thank you so much! @svekars
